src/GuiBase/Timeline/TimelineScrollArea.cpp

```diff
@@ -0,0 +1,189 @@
+#include <GuiBase/Timeline/TimelineScrollArea.h>
+
+#include <cmath>
+
+#include <QScrollBar>
+#include <QWheelEvent>
+
+#include <GuiBase/Timeline/Timeline.h>
+#include <GuiBase/Timeline/TimelineFrameSelector.h>
+
+namespace Ra::GuiBase {
+
+TimelineScrollArea::TimelineScrollArea( QWidget* parent ) : QScrollArea( parent ) {
+    horizontalScrollBar()->setStyleSheet( "\
+                    QScrollBar:horizontal {\
+                        background: transparent;\
+                    }\
+                    \
+                    QScrollBar::handle:horizontal {\
+                        background: white;\
+                    }\
+                    \
+                    QScrollBar::add-line:horizontal {\
+                        border: none;\
+                        background: transparent;\
+                    }\
+                    \
+                    QScrollBar::sub-line:horizontal {\
+                        border: none;\
+                        background: transparent;\
+                    }" );
+}
+
+int TimelineScrollArea::getNbInterval() {
+    return m_nbInterval;
+}
+
+Scalar TimelineScrollArea::getMaxDuration() {
+    return m_maxDuration;
+}
+
+void TimelineScrollArea::setMaxDuration( Scalar duration ) {
+    m_maxDuration = duration;
+}
+
+int TimelineScrollArea::getZero() {
+    return m_zero;
+}
+
+Scalar TimelineScrollArea::getPixPerSec() {
+    return m_pixPerSec;
+}
+
+Scalar TimelineScrollArea::getStep() {
+    return m_step;
+}
+
+void TimelineScrollArea::onDrawRuler( int width ) {
+    int iStep = 0;
+    while ( iStep < s_nbSteps && width * s_steps[iStep] < 50 * m_maxDuration )
+        iStep++;
+
+    if ( iStep == s_nbSteps ) { return; }
+
+    m_step = s_steps[iStep];
+    emit stepChanged( m_step );
+
+    m_nbInterval = int( std::ceil( m_maxDuration / m_step ) ) + 2;
+    m_pixPerSec  = ( Scalar( width ) / m_nbInterval ) / m_step;
+
+    m_zero = int( m_pixPerSec * m_step );
+    widget()->setMinimumWidth( width );
+
+    update();
+}
+
+void TimelineScrollArea::keyPressEvent( QKeyEvent* event ) {
+    switch ( event->key() )
```

Comment:
Is there a documentation saying which key does what ?
Again, if the user has to read the code or the reference documentation, it will not (and not use the system).

Reply:
That's on the HelpDialog sheet, which can be opened from the `?` button on the timeline ui